### PR TITLE
HP-1880: billing VPS traf init

### PR DESCRIPTION
**Pull request by bladeroot**
## Summary by CodeRabbit

- **New Features**
	- Updated VPS configuration to include new traffic-related columns (`vps_traf`, `vps_traf_in`, and `vps_traf_max`).
	- Enhanced sorting options to include `vps_traf_max`.
	- Improved resource conversion to support new VPS traffic types.
	- Added new formatting options for displaying VPS traffic information.
	- Introduced new decorators for detailed VPS traffic resource handling.

### CODE REVIEW

File: src/logic/bill/QuantityFormatterFactory.php [hunk 1]
```diff
@@ -38,6 +38,7 @@ final class QuantityFormatterFactory implements QuantityFormatterFactoryInterfac
         'devops_support'    => SupportTimeQuantity::class,
         'referral'          => DefaultQuantityFormatter::class,
         'server_traf_max'   => DefaultQuantityFormatter::class,
+        'cdn_traf_max'      => DefaultQuantityFormatter::class,
```
Comment (SilverFire) [inline src/logic/bill/QuantityFormatterFactory.php:41]:
```suggestion
        'vps_traf_max'      => DefaultQuantityFormatter::class,
```

Typo?

File: src/models/decorators/target/VpsTrafficMaxResourceDecorator.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * @package   hipanel-module-finance
+ * @license   BSD-3-Clause
+ * @copyright Copyright (c) 2015-2019, HiQDev (http://hiqdev.com/)
+ */
+
+namespace hipanel\modules\finance\models\decorators\target;
+
+use Yii;
+
+class CdnTrafficMaxResourceDecorator extends VpsTrafficResourceDecorator
```
[the remark this answers is not in the document]
Comment (SilverFire) [reply src/models/decorators/target/VpsTrafficMaxResourceDecorator.php:15]: ++

File: src/models/decorators/target/TargetResourceDecoratorFactory.php [hunk 3]
```diff
@@ -30,6 +30,9 @@ protected static function typeMap(): array
             'cdn_cache95' => CdnCache95ResourceDecorator::class,
             'cdn_traf95' => CdnTraffic95ResourceDecorator::class,
             'cdn_traf95_max' => CdnTraffic95MaxResourceDecorator::class,
+            'vps_traf' => VpsTrafficesourceDecorator::class,
+            'vps_traf_ssl' => VpsTrafficInResourceDecorator::class,
+            'vps_traf_max' => VpsTrafficMaxResourceDecorator::class,
```
[the remark this answers is not in the document]
Comment (SilverFire) [reply src/models/decorators/target/TargetResourceDecoratorFactory.php:35]: ++

File: src/models/decorators/target/VpsTrafficResourceDecorator.php [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<?php
```
Comment (SilverFire) [inline src/models/decorators/target/VpsTrafficResourceDecorator.php:1]: declare strict

File: src/models/decorators/target/VpsTrafficResourceDecorator.php [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+ */
+
+namespace hipanel\modules\finance\models\decorators\target;
+
+use hipanel\modules\finance\models\decorators\server\TrafficResourceDecorator;
+use Yii;
+
+class VpsTrafficResourceDecorator extends TrafficResourceDecorator
+{
+    public function displayTitle()
```
Comment (SilverFire) [inline src/models/decorators/target/VpsTrafficResourceDecorator.php:18]: return type

File: src/models/decorators/target/VpsTrafficInResourceDecorator.php [hunk 6]
```diff
@@ -0,0 +1,22 @@
+<?php
```
Comment (SilverFire) [inline src/models/decorators/target/VpsTrafficInResourceDecorator.php:1]: declare

File: src/models/decorators/target/VpsTrafficMaxResourceDecorator.php [hunk 7]
```diff
@@ -0,0 +1,21 @@
+<?php
+/**
```
Comment (SilverFire) [inline src/models/decorators/target/VpsTrafficMaxResourceDecorator.php:2]: declare

**Review by SilverFire — CHANGES_REQUESTED** [review]

File: src/models/decorators/target/TargetResourceDecoratorFactory.php [hunk 8]
```diff
@@ -30,6 +30,9 @@ protected static function typeMap(): array
             'cdn_cache95' => CdnCache95ResourceDecorator::class,
             'cdn_traf95' => CdnTraffic95ResourceDecorator::class,
             'cdn_traf95_max' => CdnTraffic95MaxResourceDecorator::class,
+            'vps_traf' => VpsTrafficesourceDecorator::class,
```
Comment (SilverFire) [inline src/models/decorators/target/TargetResourceDecoratorFactory.php:33]: Typo in class name

Comment (bladeroot) [reply src/models/decorators/target/TargetResourceDecoratorFactory.php:33]: ++

**Review by SilverFire — APPROVED** [review]

**Pull request merged**